src/openapi/streaming/streaming.ts

```diff
@@ -0,0 +1,1124 @@
+     *
+     * Starts in an Initializing state, transitions to Started when the Connection starts
+     * then follows the Connection state model.
+     */
+    private init() {
+        // cleanup old connection if any
+        if (this.connection) {
+            this.connection.dispose();
+        }
+
+        this.connection = new Connection(
+            this.connectionOptions,
+            this.baseUrl,
+            this.onStreamingFailed.bind(this),
+        );
+
+        this.connection.setStateChangedCallback(
+            this.onConnectionStateChanged.bind(this),
+        );
+        this.connection.setUnauthorizedCallback(this.onUnauthorized.bind(this));
+        this.connection.setReceivedCallback(this.onReceived.bind(this));
+        this.connection.setConnectionSlowCallback(
+            this.onConnectionSlow.bind(this),
+        );
+
+        // start the connection process
+        this.connect();
+    }
+
+    private onStreamingFailed() {
+        // Allow consumer to reset streaming and reconnect with different streaming service
+        this.connectionState = this.CONNECTION_STATE_DISCONNECTED;
+
+        // Let consumer setup event handlers in case of steaming failure during initial setup
+        setTimeout(() => {
+            this.trigger(this.EVENT_STREAMING_FAILED);
+        });
+    }
+
+    /**
+     * The streaming connection received a unauthorized - the token is
+     * being rejected so we should get a new one.
+     */
+    private onUnauthorized() {
+        this.authProvider.tokenRejected();
+    }
+
+    /**
+     * Reconnects the streaming socket when it is disconnected
+     */
+    private connect(isReconnection?: boolean) {
+        if (
+            this.connectionState !== this.CONNECTION_STATE_DISCONNECTED &&
+            this.connectionState !== this.CONNECTION_STATE_INITIALIZING
+        ) {
+            log.warn(
+                LOG_AREA,
+                'Only call connect on a disconnected streaming connection',
+                new Error(),
+            );
+            return;
+        }
+
+        const startConnection = () => {
+            this.setNewContextId();
+            this.updateConnectionQuery();
+
+            this.connection.start(this.onConnectionStarted.bind(this));
+        };
+
+        const expiry = this.authProvider.getExpiry();
+        if (expiry < Date.now()) {
+            // in case the refresh timer has disappeared, ensure authProvider is
+            // fetching a new token
+            const transport = this.getActiveTransportName();
+            this.authProvider.refreshOpenApiToken();
+            this.authProvider.one(
+                this.authProvider.EVENT_TOKEN_RECEIVED,
+                () => {
+                    // FIXME looks like this block is currently never executed since
+                    // isReconnection arg is never provided
+                    if (isReconnection && !this.reconnecting) {

```

Comment:
Fix me.

Reply:
Checked and `isReconnection` is provided in one case. Will remove the comment.